lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -865,3 +865,79 @@ def test_course_catalog_access_num_queries(self, user_attr_name, action, course_
         course_overview = CourseOverview.get_from_id(course.id)
         with self.assertNumQueries(num_queries, table_blacklist=QUERY_COUNT_TABLE_BLACKLIST):
             bool(access.has_access(user, action, course_overview, course_key=course.id))
+
+
+@ddt.ddt
+class CCXTestCase(SharedModuleStoreTestCase):
+    """
+    Tests for ccx course visibility
+    """
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    def setUp(self):
+        """
+        Set up tests
+        """
+        super(CCXTestCase, self).setUp()
+
+        course = CourseFactory.create(enable_ccx=True)
+        CourseOverview.load_from_module_store(course.id)
+        self.course = course
+
+        student = UserFactory.create(is_staff=False)
+        staff = UserFactory.create(is_staff=True)
+        coach = AdminFactory.create()
```

Comment:
FYI: [`AdminFactory`](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/tests/factories.py#L120-L121) is just a `UserFactory` with `is_staff = True`.  And CCX Coaches don't have to have `is_staff=True`, they can be normal users.  It's the `coach` permission granted on the CCX base course that gives them special powers.

So, could you change the coach to be a normal `UserFactory.create()`, so we know for sure that we're testing a normal CCX coach instead of a staff + ccx coach user?

This will flip the `('coach', 'enroll', True),` check to `('coach', 'enroll', False),`.

Reply:
done :)